### PR TITLE
[TT-9504] Adding TraceID to response headers

### PR DESCRIPTION
**Pull request by mativm02**
## Description
We're returning the TraceID in the response headers if OpenTelemetry is enabled to make debugging easier.

## Related Issue
https://tyktech.atlassian.net/jira/software/c/projects/TT/boards/42?modal=detail&selectedIssue=TT-9504

## Motivation and Context
https://tyktech.atlassian.net/jira/software/c/projects/TT/boards/42?modal=detail&selectedIssue=TT-9504

## How This Has Been Tested
Manual Testing:
### Status 504
Request in Postman (note the `X-Tyk-Trace-Id` header):
image
How it looks in Jaeger (you can check the Trace ID in the URL path, or next to the trace name):
image

### Status 200
Request in Postman (note the `X-Tyk-Trace-Id` header):
image
How it looks in Jaeger (you can check the Trace ID in the URL path, or next to the trace name):
image

### Status 404
Request in Postman (note the `X-Tyk-Trace-Id` header):
image
How it looks in Jaeger (you can check the Trace ID in the URL path, or next to the trace name):
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

### CODE REVIEW

**Review by tbuchaillot — CHANGES_REQUESTED** [review]
@mativm02 a couple of things:
- Unit tests missing
- This will only work for success requests 
- Is the `X-Otel-Trace-Id` expected or should we follow `w3c` conventions?

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5385/merge
Commit: 719219c998154e5f19483fc0aa8055d09d293ab1 
Triggered by: pull_request (@mativm02)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5762748293)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5385/merge
Commit: 038996e0c492184b305d73bde009eb37d1e4647c 
Triggered by: pull_request (@mativm02)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5797143900)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5385/merge
Commit: 5580e5fc1bccd74a0d0cceb6ba90742e5f5dfd5a 
Triggered by: pull_request (@mativm02)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5798814877)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5385/merge
Commit: 1181fedf775e7e8b4a02a3c069292f7e0bbc0e39 
Triggered by: pull_request (@mativm02)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5808986408)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5385/merge
Commit: 1364079414929a187f78092df7be91b4ab6335e9 
Triggered by: pull_request (@mativm02)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5809095401)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5385/merge
Commit: cabaee5b2bb0f5fac73ec929e36679c11e6d3340 
Triggered by: pull_request (@mativm02)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5809164699)

File: gateway/handler_success.go [hunk 1]
```diff
@@ -76,6 +77,15 @@ func addVersionHeader(w http.ResponseWriter, r *http.Request, globalConf config.
 	}
 }
 
+func addTraceID(w http.ResponseWriter, r *http.Request, globalConf config.Config) {
```
Comment (tbuchaillot) [inline gateway/handler_success.go:80]: I'd change this function signature to receive if Opentelemetry is enabled/disabled and move the whole function into `internal/otel` pkg

Comment (mativm02) [reply gateway/handler_success.go:80]: @tbuchaillot makes total sense to do it instead of sending the whole config object, good catch! 👌

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5385/merge
Commit: 76f7291aaa8bc4d739b5f73bff2ea0695fb6a504 
Triggered by: pull_request (@mativm02)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5822406397)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5385/merge
Commit: 0fcdf9207643307b65ffdb5b850eddb37767baf9 
Triggered by: pull_request (@mativm02)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5822399577)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5385/merge
Commit: 6ad3157def7629bce9ebe167012971584ae811cb 
Triggered by: pull_request (@mativm02)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5832503538)

**Review by tbuchaillot — APPROVED** [review]
LGTM

**Pull request merged**